snippets/csharp/VS_Snippets_Wpf/DrawingMiscSnippets_snip/CSharp/EnumerateDrawingsExample.xaml.cs

```diff
@@ -5,65 +5,59 @@
         {
             RetrieveDrawing(MainBorder);
         }
 
         //<SnippetGraphicsMMRetrieveDrawings>
         public void RetrieveDrawing(Visual v)
         {
-            DrawingGroup dGroup = VisualTreeHelper.GetDrawing(v);
-            EnumDrawingGroup(dGroup);
-
+            DrawingGroup drawingGroup = VisualTreeHelper.GetDrawing(v);
+            EnumDrawingGroup(drawingGroup);
         }
 
-         // Enumerate the drawings in the DrawingGroup.
-         public void EnumDrawingGroup(DrawingGroup drawingGroup)
-         {
-             DrawingCollection dc = drawingGroup.Children;
-
-             // Enumerate the drawings in the DrawingCollection.
-             foreach (Drawing drawing in dc)
-             {
-                 // If the drawing is a DrawingGroup, call the function recursively.
-                 if (drawing.GetType() == typeof(DrawingGroup))
-                 {
-                     EnumDrawingGroup((DrawingGroup)drawing);
-                 }
-                 else if (drawing.GetType() == typeof(GeometryDrawing))
-                 {
-                     // Perform action based on drawing type.  
-                 }
-                 else if (drawing.GetType() == typeof(ImageDrawing))
-                 {
-                     // Perform action based on drawing type.
-                 }
-                 else if (drawing.GetType() == typeof(GlyphRunDrawing))
-                 {
-                     // Perform action based on drawing type.
-                 }
-                 else if (drawing.GetType() == typeof(VideoDrawing))
-                 {
-                     // Perform action based on drawing type.
-                 }
-             }
-         }
-         //</SnippetGraphicsMMRetrieveDrawings>
-     }
-    
+        // Enumerate the drawings in the DrawingGroup.
+        public void EnumDrawingGroup(DrawingGroup drawingGroup)
+        {
+            DrawingCollection dc = drawingGroup.Children;
+
+            // Enumerate the drawings in the DrawingCollection.
+            foreach (Drawing drawing in dc)
+            {
+                // If the drawing is a DrawingGroup, call the function recursively.
+                if (drawing is DrawingGroup @group)
+                {
+                    EnumDrawingGroup(@group);
+                }
+                else if (drawing is GeometryDrawing)
+                {
+                    // Perform action based on drawing type.  
+                }
+                else if (drawing is ImageDrawing)
+                {
+                    // Perform action based on drawing type.
+                }
+                else if (drawing is GlyphRunDrawing)
+                {
+                    // Perform action based on drawing type.
+                }
+                else if (drawing is VideoDrawing)
+                {
+                    // Perform action based on drawing type.
+                }
```

Comment:
There are just enough combinations here to consider a pattern matching switch instead of the `is` statement.

What do you think @lindexi ?

Reply:
@BillWagner I also think `switch` is better than `is` but the original code has not used the converted value. I afraid my changed is too large and I am worried about modifying the code structure that I only change the `GetType` to `is`.

Reply:
@BillWagner Do you think it can be pattern matching switch?

Reply:
@lindexi Either one works well. So, let's keep it as is.

I'll :shipit: now. 

Thanks for contributing to docs. 🎆